### PR TITLE
Mila/count add api interface

### PR DESCRIPTION
**Pull request by milaGGL**
This count query interface is implemented based on the [COUNT Query API proposal]([go/firestore-count-api-client-proposal](http://goto.google.com/firestore-count-api-client-proposal)).

Based on the demo provided: 

```
async function Demo1_CountOfDocumentsInACollection(db: Firestore) {
  const countQuery_ = countQuery(collection(db, "games/halo/players"));
  const snapshot = await getAggregateFromServerDirect(countQuery_);
  expect(snapshot.getCount()).to.equal(5000000);
}

async function Demo2_CountOfDocumentsInACollectionWithFilter(db: Firestore) {
  const query_ = query(collection(db, "games/halo/players"), where("online", "==", true));
  const countQuery_ = countQuery(query_);
  const snapshot = await getAggregateFromServerDirect(countQuery_);
  expect(snapshot.getCount()).to.equal(2000);
}
```

- [ ] `countQuery` function will accept a collection or query reference
- [ ]  `countQuery `will create a `aggregateQuery` instance, and return it. (class AggregateQuery)
- [ ] `getAggregateFromServerDirect` function will take aggregateQuery object, create `AggregateQuerySnapshot` instance, and return a promise
- [ ] `AggregateQuerySnapshot` object has `getCount` public method to access the document count

### CODE REVIEW

File: common/api-review/firestore.api.md [hunk 1]
```diff
@@ -17,6 +17,30 @@ export type AddPrefixToKeys<Prefix extends string, T extends Record<string, unkn
     [K in keyof T & string as `${Prefix}.${K}`]+?: T[K];
 };
```
Comment (milaGGL) [inline common/api-review/firestore.api.md:19]: auto-created by API Extractor

File: packages/firestore/src/lite-api/aggregate.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ import {DocumentData, Query, queryEqual} from './reference';
+
+
+ export class AggregateQuery<T = DocumentData>{
```
Comment (milaGGL) [inline packages/firestore/src/lite-api/aggregate.ts:21]: created class only, hasn't implement the internal methods

File: packages/firestore/src/lite-api/aggregate.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+export function countQuery(query: Query): AggregateQuery {
+  return new AggregateQuery(query);
+}
+
+export function getAggregateFromServerDirect(
+  query: AggregateQuery
+): Promise<AggregateQuerySnapshot> {
+  return Promise.resolve(new AggregateQuerySnapshot(query, 42));
```
Comment (milaGGL) [inline packages/firestore/src/lite-api/aggregate.ts:51]: 42 here is a dummy count

File: packages/firestore/test/integration/api/count.test.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+ */
+
+import { expect } from 'chai';
+import {
+  countQuery,
+  getAggregateFromServerDirect,
+  query
+} from '../util/firebase_export';
+import { apiDescribe, withTestCollection } from '../util/helpers';
+
```
Comment (milaGGL) [inline packages/firestore/test/integration/api/count.test.ts:25]: both tests will fail as the count API internal logic is not implemented yet, and only returning 42 as dummy count.

**Review by dconeybe — APPROVED** [review]

**Pull request merged**